File: common/metadata.py
from datetime import datetime


def create_prefix(metadata):

    source = metadata.get(
        "source",
        "unknown"
    )

    city = metadata.get(
        "city",
        "unknown"
    )

    country = metadata.get(
        "country",
        ""
    )

    scraped_at = metadata.get(
        "scrapedAt"
    )

    timestamp = (
        datetime
        .fromisoformat(
            scraped_at.replace(
                "Z",
                "+00:00"
            )
        )
        .strftime(
            "%Y%m%dT%H%M%S"
        )
    )


    parts = [
        source,
        city,
        country,
        timestamp
    ]

    return "_".join(
        x for x in parts if x
    )

File: common/metadata.js
function formatTimestamp(scrapedAt) {
    return new Date(scrapedAt)
        .toISOString()
        .replace(/[-:]/g, "")
        .replace(/\.\d{3}Z$/, "");
}

function createPrefix(metadata) {
    const timestamp =
        formatTimestamp(metadata.scrapedAt);

    return [
        metadata.source,
        metadata.city,
        metadata.country,
        timestamp
    ]
    .filter(Boolean)
    .join("_");
}

module.exports = {
    createPrefix
};